Library/Homebrew/dependency.rb

```diff
@@ -29,7 +29,31 @@ def hash
   end
 
   def to_formula
-    formula = Formulary.factory(name)
```

Comment:
I don't think you should change this method. There are more underlying problems here regarding dependency resolution. (See https://github.com/Homebrew/homebrew/pull/40991#issuecomment-116948161 and https://github.com/Homebrew/homebrew/issues/14089#issuecomment-39590767).

What you need to do is updating method [`TapLoader#initialize `](https://github.com/Homebrew/homebrew/blob/9124253ae9866684f171bffebd761714d68d2a0f/Library/Homebrew/formulary.rb#L145) and [`Formulary#load_ref` (after searching alias)](https://github.com/Homebrew/homebrew/blob/9124253ae9866684f171bffebd761714d68d2a0f/Library/Homebrew/formulary.rb#L228).  The idea is to trying old name if formula is not found when invoking `Formulary#factory`.